react/OrderItems.tsx

```diff
@@ -665,7 +685,8 @@ const OrderItemsProvider: FC = ({ children }) => {
     (
       items: Array<Partial<CatalogItem>>,
       marketingData?: Partial<MarketingData>,
-      salesChannel?: string
+      salesChannel?: string,
+      options?: AddItemOptions
```

Comment:
this doesn't look good as an API when we want to send some of these options but don't want to set the `salesChannel`, resulting in an awkward call:

```js
addItem(
  itemsToAdd,
  {},
  null,
  { allowOutdatedData: ['paymentData'] }
)
```

I think we could transform this `salesChannel` argument into "two" arguments, like so

```ts
const addItem = (
  items: Array<Partial<CatalogItem>>,
  marketingData?: Partial<MarketingData>,
  salesChannelOrOptions?: string | AddItemOptions
) => {
  const salesChannel = typeof salesChannelOrOptions === 'string'
    ? salesChannelOrOptions
    : salesChannelOrOptions.salesChannel
  const options = typeof salesChannelOrOptions !== 'string'
    ? salesChannelOrOptions
    : {}

  // ...
}
```

what do you think? this way we have a nicer API and still wouldn't be a breaking change

Reply:
This seems weirder to me because it creates a dichotomy between `salesChannel` and `options` that doesn't seem to make sense.
In my opinion, `salesChannel` should be required (its value is always needed, right? Even when it is `1`) **OR** it should be part of `options`.

Reply:
@darlenedms I thought of this so we wouldn't introduce a breaking change, `salesChannel` isn't required in this route and making it required would mean we would need to also make `marketingData` required as well (since we can't have a required argument after an optional one)

Reply:
it would've been better if we unified the `marketingData`, `salesChannel` and `allowedOutdatedData` arguments into one `options` argument, but we can't go back and change the `marketingData` and `salesChannel` args at this point :-/

Reply:
I prefer the solution proposed by @lucasecdb more, but it still doesn't look like a great API. How about a third option:

1. create a new function like `pushItem`
1. make old `addItem` use `pushItem`
1. deprecate `addItem`

```typescript
const pushItem = (
  items: Array<Partial<CatalogItem>>,
  options?: AddItemOptions
) => {
  // ...
}

const addItem = (
  items: Array<Partial<CatalogItem>>,
  marketingData?: Partial<MarketingData>,
  salesChannel?: string
) => {
  return pushItem(items, { marketingData, salesChannel })
}
```

That way, we can make the API the best that we can. The only thing that bothers me is having to rename the function. What do you people think?

Reply:
Nice, @wdsrocha! And we wait for the next breaking change to get our `addItem` name back. 😈

Reply:
nitpick, what about `addItems` (plural) instead of `pushItem`?

Reply:
Great, @lucasecdb!